### PR TITLE
fix(worker): fixed the fcm data message issue with payload messed with additional data

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

The FCM data message `data` field should be a type of `Record<string, string>` otherwise the FCM platform will raise an error.

FCM docs:
https://firebase.google.com/docs/cloud-messaging/concept-options

https://firebase.google.com/docs/reference/admin/node/firebase-admin.messaging.basemessage.md#basemessage_interface

### Other information (Screenshots)

Screenshot 2023-07-08 at 00 31 19
Screenshot 2023-07-08 at 00 31 08

### CODE REVIEW

File: apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts [hunk 1]
```diff
@@ -147,7 +149,17 @@ export class SendMessagePush extends SendMessageBase {
...
 
-      await this.sendMessage(integration, channel.credentials.deviceTokens, title, content, command, data, overrides);
+      await this.sendMessage(
+        subscriber,
+        integration,
+        channel.credentials.deviceTokens,
+        title,
+        content,
+        command,
+        command.payload,
```
Comment (LetItRock) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts:159]: this line basically is the fix... the payload was containing additional data that shouldn't be sent with the FCM data message

File: apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts [hunk 2]
```diff
@@ -217,6 +231,8 @@ export class SendMessagePush extends SendMessageBase {
         content,
         payload,
         overrides,
+        subscriber,
+        step,
```
Comment (LetItRock) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts:235]: this data is needed only for the `webhook` provider, so I'm sending them as a separate fields

File: providers/fcm/src/lib/fcm.provider.spec.ts [hunk 3]
```diff
@@ -257,3 +277,49 @@ test('should trigger fcm data for android with priority option', async () => {
     },
   });
 });
+
+test('should clean the payload for the FCM data message', async () => {
```
Comment (LetItRock) [inline providers/fcm/src/lib/fcm.provider.spec.ts:281]: test that makes sure that the FCM data message data field is `Record<string, string>`

File: providers/fcm/src/lib/fcm.provider.ts [hunk 4]
```diff
@@ -38,17 +38,18 @@ export class FcmPushProvider implements IPushProvider {
...
   }
 
   async sendMessage(
-    options: IPushOptions
+    options: IPushOptions,
   ): Promise<ISendMessageSuccessResponse> {
     delete (options.overrides as { deviceTokens?: string[] })?.deviceTokens;
 
     const overridesData = options.overrides || ({} as any);
+    const payload = this.cleanPayload(options.payload);
```
Comment (LetItRock) [inline providers/fcm/src/lib/fcm.provider.ts:52]: clean the payload, like transform it to `Record<string, string>`

File: providers/push-webhook/src/lib/push-webhook.provider.ts [hunk 5]
```diff
@@ -15,13 +15,20 @@ export class PushWebhookPushProvider implements IPushProvider {
...
+    options: IPushOptions,
   ): Promise<ISendMessageSuccessResponse> {
-    const bodyData = this.createBody(options);
+    const bodyData = this.createBody({
+      ...options,
+      payload: {
+        ...options.payload,
+        subscriber: options.subscriber,
+        step: options.step,
+      },
```
Comment (LetItRock) [inline providers/push-webhook/src/lib/push-webhook.provider.ts:30]: for the `webhook` provider send `subscriber, step` as the payload, like we did before

File: apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts [hunk 6]
```diff
@@ -59,13 +60,14 @@ export class SendMessagePush extends SendMessageBase {
 
     const pushChannel: NotificationStepEntity = command.step;
 
+    const stepData: IPushOptions['step'] = {
```
Comment (p-fernandez) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts:63]: I like this separation. You could also call the object `step` already.

File: packages/stateless/src/lib/provider/provider.interface.ts [hunk 7]
```diff
@@ -68,6 +68,12 @@ export interface IPushOptions {
     };
     fcmOptions?: { analyticsLabel?: string };
   };
+  subscriber: object;
+  step: {
+    digest: boolean;
+    events: object[] | undefined;
+    total_count: number | undefined;
+  };
```
Comment (p-fernandez) [inline packages/stateless/src/lib/provider/provider.interface.ts:76]: Can't here we reuse `IPushOptions['step']`? If it is needed let's move it to `@novu/shared`.

File: providers/fcm/src/lib/fcm.provider.ts [hunk 8]
```diff
@@ -93,7 +94,7 @@ export class FcmPushProvider implements IPushProvider {
       throw new Error(
         `Sending message failed due to "${
           res.responses.find((i) => i.success === false).error.message
-        }"`
+        }"`,
```
Comment (p-fernandez) [inline providers/fcm/src/lib/fcm.provider.ts:97]: I think you might have a change of configuration in your prettifying.

File: apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts (hunk 6 above)
Comment (LetItRock) [reply apps/worker/src/app/workflow/usecases/send-message/send-message-push.usecase.ts:63]: I didn't want to confuse it with a real `step`

File: packages/stateless/src/lib/provider/provider.interface.ts (hunk 7 above)
Comment (LetItRock) [reply packages/stateless/src/lib/provider/provider.interface.ts:76]: this is the `IPushOptions` interface :P

File: providers/fcm/src/lib/fcm.provider.ts (hunk 8 above)
Comment (p-fernandez) [reply providers/fcm/src/lib/fcm.provider.ts:97]: We need to disable it. But I have no access to it.

File: packages/stateless/src/lib/provider/provider.interface.ts (hunk 7 above)
Comment (p-fernandez) [reply packages/stateless/src/lib/provider/provider.interface.ts:76]: You got my point 😂

**Review by p-fernandez — APPROVED** [review]
🌟

**Pull request merged**